src/ppx/grfx/vk/vk_command.cpp

```diff
@@ -1035,6 +1035,65 @@ void CommandBuffer::CopyImageToImage(
         &region);
 }
 
+void CommandBuffer::BlitImage(
+    const grfx::ImageBlitInfo* pCopyInfo,
+    grfx::Image*               pSrcImage,
+    grfx::Image*               pDstImage)
+{
+    bool isSourceDepthStencil = grfx::GetFormatDescription(pSrcImage->GetFormat())->aspect == grfx::FORMAT_ASPECT_DEPTH_STENCIL;
+    bool isDestDepthStencil   = grfx::GetFormatDescription(pDstImage->GetFormat())->aspect == grfx::FORMAT_ASPECT_DEPTH_STENCIL;
+    PPX_ASSERT_MSG(isSourceDepthStencil == isDestDepthStencil, "both images in an image copy must be depth-stencil if one is depth-stencil");
```

Comment:
Is it fine for them to differ if they are not both depth stencil? For example, one depth, one color

Reply:
I belive so, the spec only says that  if one has depth or stencil the other one should also have it. No other restiction on the format

Reply:
I see from the spec "If either of srcImage or dstImage was created with a depth/stencil format, the other must have exactly the same format", which is not what we are checking now. Also it seems like if formats are one with just depth and one with just stencil the assert will not trigger?

Reply:
Yes, you are right I see your point. LI guess the correct check will be that both (dstImg and srcImg) have the same value in both fields (depth and stencil)

Reply:
Actually, I switched it again to match what I believe the spec says.

Reply:
I think it should be depth or stencil, maybe smth like this `isSourceDepthOrStencil = grfx::GetFormatDescription(pSrcImage->GetFormat())->aspect & grfx::FORMAT_ASPECT_DEPTH_STENCIL;` so it is true if it has either depth or stencil or both
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/5268c4004524149b6bb7230969038e834914d652/layers/core_checks/cc_copy_blit_resolve.cpp#L2937
Now that I am looking at the other code, I am not sure if this was copied from the other function. By the way, there are a bunch of other rules for using the function, so I am not sure if we need to assert this specific one. Although might still be useful.